### DISCUSSION TITLE
Unable to link to Heroku

### DISCUSSION THREAD

**Discussion by CalTech1200** [FoxxMD/context-mod#75] [Q&A]
Hello, I received the following error:

> -----> Building on the Heroku-20 stack
> -----> Determining which buildpack to use for this app
> -----> Node.js app detected
>        
> -----> Creating runtime environment
>        
>        NPM_CONFIG_LOGLEVEL=error
>        NODE_VERBOSE=false
>        NODE_ENV=production
>        NODE_MODULES_CACHE=true
>        
> -----> Installing binaries
>        engines.node (package.json):  >=15
>        engines.npm (package.json):   unspecified (use default)
>        
>        Resolving node version >=15...
>        Downloading and installing node 17.4.0...
>        Using default npm version: 8.3.1
>        
> -----> Installing dependencies
>        Installing node modules
>        npm ERR! code EINTEGRITY
>        npm ERR! sha512-OnunfctuocKBmG0uzkBkVYGSW13VYmzglQPwj4ZFOBEtV2e2ECqe65eCaYRYHSvKynGc8T5hiaEWAVR2hVj+Yg== integrity checksum failed when using sha512: wanted sha512-OnunfctuocKBmG0uzkBkVYGSW13VYmzglQPwj4ZFOBEtV2e2ECqe65eCaYRYHSvKynGc8T5hiaEWAVR2hVj+Yg== but got sha512-sQFPdrnss4TJ65YKSFFnK7LZadVSslv9wPUgSOZRrcTW41oMU0wl11adiRUizAPHHWx+h0S2KMTNZgiSKa+3pg==. (34955 bytes)
>        
>        npm ERR! A complete log of this run can be found in:
>        npm ERR!     /tmp/npmcache.v3hNx/_logs/2022-02-06T01_58_54_506Z-debug-0.log
> -----> Build failed
>        
>        We're sorry this build is failing! You can troubleshoot common issues here:
>        https://devcenter.[heroku.com](http://heroku.com/home)/articles/troubleshooting-node-deploys
>        
>        Some possible problems:
>        
>        - Dangerous semver range (>) in engines.node
>          https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
>        
>        Love,
>        Heroku
>        
>  !     Push rejected, failed to compile Node.js app.
>  !     Push failed
> Build finished

**Comment by FoxxMD** [comment]
This is because I use a fork of [winston](https://github.com/FoxxMD/winston/tree/customChildMerge) to address some issues with [child meta object merging](https://github.com/winstonjs/winston/issues/2029) that has not be resolved in the main branch. There are [PRs for resolving](https://github.com/winstonjs/winston/pull/1989) this [open](https://github.com/winstonjs/winston/pull/2038) but I'm not sure when/if they will be merged.

Another user had this issue on their local machine and was able to resolve it by [clearing the NPM cache](https://discord.com/channels/921051902576304170/921051903062859828/921127754026586162) but I don't think that's feasible for heroku since you don't have control over the installation process.

I don't have the bandwidth to try to fix heroku right now and this issue will eventually resolve its self when the main winston branch gets its ducks in a row. I'd suggest trying to use the docker build instead or run the project locally for now.
